protocol_connect.go

```diff
@@ -478,6 +479,7 @@ func (cc *connectStreamingClientConn) Receive(msg any) error {
 		// For users to realize that the stream has ended, Receive must return an
 		// error.
 		serverErr.meta = cc.responseHeader.Clone()
+		serverErr.isServerErr = true
```

Comment:
I think we can avoid both this change and the one above by handling this closer to the wire - let's add this code to `connectWireError.asError` instead.